ls1intum/Artemis
# `Programming exercises`: Include practice participations in the participation table

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


Instructors should be able to see the practice participations after the deadline without affecting the scores or the assessment of the exercise.

### Description

The practice participation are now included in the participation overview of the exercise. 
Additionally there is a filter to only show the graded participations. The filter is only visible after the deadline, since there cannot be any practice participations before the deadline.
The practice participations should not be visible in the submissions, the scores or the assessment view of the programming exercise.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Programming Exercise with a deadline

1. Log in to Artemis.
2. Participate with each student before the deadline.
3. Participate after the deadline using the practice mode after the deadline.
4. Check during each step if the participation overview looks correct to the instructor.
5. Check that submissions, scores and assessments only show the graded participation.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| ParticipationService.java | 78% | 91% |
| ParticipationResource.java | 66% | 90% |
